apps/admin-ui/Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM node:18-bullseye AS base
+
+FROM base AS builder
+WORKDIR /app
+RUN npm install -g turbo
+COPY . .
+RUN turbo prune --scope=admin-ui --docker
+
+# Add lockfile and package.json's of isolated subworkspace
+FROM base AS installer
+WORKDIR /app
+# First install the dependencies (as they change less often)
+COPY .gitignore .gitignore
+COPY --from=builder /app/out/json/ .
+COPY --from=builder /app/out/pnpm-lock.yaml ./pnpm-lock.yaml
+RUN corepack enable
+RUN pnpm install --frozen-lockfile
+
+# Build the project
+COPY --from=builder /app/out/full/ .
+COPY turbo.json turbo.json
+ENV SKIP_ENV_VALIDATION=true
+RUN pnpm turbo run build --filter=admin-ui...
+
+FROM base AS runner
+WORKDIR /app
+
+# Don't run production as root
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
```

Comment:
Nitpick: These could be a single line (creates one less layer in the final image):

```dockerfile
RUN addgroup --system --gid 1001 nodejs \
    && adduser --system --uid 1001 nextjs
```